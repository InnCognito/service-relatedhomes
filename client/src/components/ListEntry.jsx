import React from 'react';

export default class ListEntry extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      img: this.props.entry.img,
      house_type: this.props.entry.house_type,
      location: this.props.entry.location,
      description: this.props.entry.description,
      cost_per_night: this.props.entry.cost_per_night,
      rating: this.props.entry.rating,
      votes: this.props.entry.votes
    }
  }

  render() {
    return (
      <div className="individualPic">
        <img className="img" src={this.state.img}/>
        <p className="location">{this.state.house_type} Â· {this.state.location}</p>
        <p className="description">{this.state.description}</p>
        <p className="price">${this.state.cost_per_night} per night</p>
        <p className="votes">{this.state.votes}</p>
      </div>
    )
  }
}
