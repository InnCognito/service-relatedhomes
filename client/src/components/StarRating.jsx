import React from 'react';

export default class StarRating extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      rating: (Math.round((this.props.rating / 5) * 100)).toString() + '%',
    };
  }

  render() {
    return (
      <div className="starsOuter">
        <div className="starsInner" style={{width: this.state.rating}}>
        </div>
      </div>
    );
  }
}
