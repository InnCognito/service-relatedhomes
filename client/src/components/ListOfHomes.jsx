import React from 'react';
import ListEntry from './ListEntry.jsx';

var ListOfHomes = ({ data }) => {
  return (
    <div>
      {data.map((entry) => <ListEntry entry={entry} key={entry.id} />)}
    </div>
  );
}

export default ListOfHomes;