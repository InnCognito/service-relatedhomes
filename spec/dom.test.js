import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ListEntry from '../client/src/components/ListEntry';
import StarRating from '../client/src/components/StarRating';
import ListOfHomes from '../client/src/components/ListOfHomes';
configure({ adapter: new Adapter() });

const testEntries = (() => {
  let arr = [];
  for (let i = 0; i < 5; i++) {
    arr.push({
      id: i,
      img: `path${i}.jpg`,
      house_type: 'house',
      location: 'Berlin',
      description: `sunny${i}`,
      cost_per_night: 250 + i,
      rating: 1.1 + (i * 0.5),
      votes: 4
    });
  }
  return arr;
})();

// snapshot
it('render ListEntry correctly', () => {
  const tree = renderer
    .create(<ListEntry entry={testEntries[0]} key={testEntries.id} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('render StarRating correctly', () => {
  const tree = renderer
    .create(<StarRating rating={testEntries[0].rating} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('render ListOfHomes correctly', () => {
  const tree = renderer
    .create(<ListOfHomes data={testEntries} height={720} width={1150} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// jest-enzyme
test('component of ListEntry', () => {
  const entry = shallow(<ListEntry entry={testEntries[1]} key={testEntries[1].id} />);
  expect(entry.text()).not.toBeNull();
});

test('simulate button click', () => {
  const wrapper = shallow(<ListOfHomes data={testEntries} height={720} width={1150} />);
  expect(wrapper.find('ListEntry').first().html()).toMatchSnapshot();
  wrapper.find('button.rightScroll').simulate('click');
  expect(wrapper.find('ListEntry').first().html()).toMatchSnapshot();
  wrapper.find('button.leftScroll').simulate('click');
  expect(wrapper.find('ListEntry').first().html()).toMatchSnapshot();
});

test('number of rendered items', () => {
  let wrapper = shallow(<ListOfHomes data={testEntries} height={720} width={1340} />);
  expect(wrapper.find(ListEntry).last().key()).toEqual("2");
  wrapper = shallow(<ListOfHomes data={testEntries} height={720} width={1000} />);
  expect(wrapper.find(ListEntry).last().key()).toEqual("1");
  wrapper = shallow(<ListOfHomes data={testEntries} height={720} width={550} />);
  expect(wrapper.find(ListEntry).last().key()).toEqual("0");
});
