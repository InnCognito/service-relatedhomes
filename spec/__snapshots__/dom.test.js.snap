// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ListEntry correctly 1`] = `
<a
  className="individualPic"
  href="/?id=0"
>
  <div>
    <img
      className="img"
      src="path0.jpg"
    />
    <p
      className="location"
    >
      house
       · 
      Berlin
    </p>
    <p
      className="description"
    >
      sunny0
    </p>
    <p
      className="price"
    >
      $
      250
       per night
    </p>
    <div
      className="starsOuter"
    >
      <div
        className="starsInner"
        style={
          Object {
            "width": "22%",
          }
        }
      />
    </div>
    <span
      className="votes"
    >
      4
    </span>
  </div>
</a>
`;

exports[`render ListOfHomes correctly 1`] = `
<div
  className="flex-container"
>
  <button
    className="leftScroll"
    onClick={[Function]}
    style={
      Object {
        "visibility": "hidden",
      }
    }
  >
    &lt;
  </button>
  <a
    className="individualPic"
    href="/?id=0"
  >
    <div>
      <img
        className="img"
        src="path0.jpg"
      />
      <p
        className="location"
      >
        house
         · 
        Berlin
      </p>
      <p
        className="description"
      >
        sunny0
      </p>
      <p
        className="price"
      >
        $
        250
         per night
      </p>
      <div
        className="starsOuter"
      >
        <div
          className="starsInner"
          style={
            Object {
              "width": "22%",
            }
          }
        />
      </div>
      <span
        className="votes"
      >
        4
      </span>
    </div>
  </a>
  <a
    className="individualPic"
    href="/?id=1"
  >
    <div>
      <img
        className="img"
        src="path1.jpg"
      />
      <p
        className="location"
      >
        house
         · 
        Berlin
      </p>
      <p
        className="description"
      >
        sunny1
      </p>
      <p
        className="price"
      >
        $
        251
         per night
      </p>
      <div
        className="starsOuter"
      >
        <div
          className="starsInner"
          style={
            Object {
              "width": "32%",
            }
          }
        />
      </div>
      <span
        className="votes"
      >
        4
      </span>
    </div>
  </a>
  <button
    className="rightScroll"
    onClick={[Function]}
    style={
      Object {
        "visibility": "visible",
      }
    }
  >
    &gt;
  </button>
</div>
`;

exports[`render StarRating correctly 1`] = `
<div
  className="starsOuter"
>
  <div
    className="starsInner"
    style={
      Object {
        "width": "22%",
      }
    }
  />
</div>
`;

exports[`simulate button click 1`] = `"<a href=\\"/?id=0\\" class=\\"individualPic\\"><div><img class=\\"img\\" src=\\"path0.jpg\\"/><p class=\\"location\\">house · Berlin</p><p class=\\"description\\">sunny0</p><p class=\\"price\\">$250 per night</p><div class=\\"starsOuter\\"><div class=\\"starsInner\\" style=\\"width:22%\\"></div></div><span class=\\"votes\\">4</span></div></a>"`;

exports[`simulate button click 2`] = `"<a href=\\"/?id=1\\" class=\\"individualPic\\"><div><img class=\\"img\\" src=\\"path1.jpg\\"/><p class=\\"location\\">house · Berlin</p><p class=\\"description\\">sunny1</p><p class=\\"price\\">$251 per night</p><div class=\\"starsOuter\\"><div class=\\"starsInner\\" style=\\"width:32%\\"></div></div><span class=\\"votes\\">4</span></div></a>"`;

exports[`simulate button click 3`] = `"<a href=\\"/?id=0\\" class=\\"individualPic\\"><div><img class=\\"img\\" src=\\"path0.jpg\\"/><p class=\\"location\\">house · Berlin</p><p class=\\"description\\">sunny0</p><p class=\\"price\\">$250 per night</p><div class=\\"starsOuter\\"><div class=\\"starsInner\\" style=\\"width:22%\\"></div></div><span class=\\"votes\\">4</span></div></a>"`;
